src/ray/util/logging.cc

```diff
@@ -312,14 +311,45 @@ void RayLog::InitLogFormat() {
   }
 }
 
+/*static*/ std::string RayLog::GetLogOutputFilename(const std::string &log_dir,
+                                                    const std::string &log_file,
+                                                    const std::string &app_name) {
+  // Case-1
```

Comment:
these Case-n are not informative. We can write a prologue like: 

```
We combine log_dir and (log_file or app_name) into a final output file name. Rules:

1. both log_dir and log_file are empty: return "" meaning no log outputs.
1. both log_dir and log_file are NON empty: return f"{log_dir}/{log_name}".
2. log_dir is NON empty, log_file is empty: return f"{log_dir}/{app_name}_{pid}.log".
3. log_dir is empty, log_file is NON empty: check failure.
```

Then the "Case-n" comments would become meaningful.

Reply:
The case-n corresponds to header file.

Reply:
Code reference here: https://github.com/ray-project/ray/blob/c49bae6d2e34e50f5fa625ddad0db1cef717a0ab/src/ray/util/logging.h#L268-L273

Reply:
ok